backend/checkout.go

```diff
@@ -0,0 +1,84 @@
+// Copyright Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+var discounts map[string]float32
+
+func InitCheckout() {
+	http.HandleFunc("/checkout/shopping-cart", handleShoppingCart)
+	http.HandleFunc("/checkout/apply-code", handleApplyCode)
+	discounts = make(map[string]float32)
+}
+
+func handleApplyCode(w http.ResponseWriter, r *http.Request) {
+	EnableCors(w, r)
+	SetContentTypeJson(w)
+	SetMaxAge(w, 0)
+	if r.Method == "POST" {
+		clientId := r.FormValue("clientId")
+		discounts[clientId] = 0.8
+		writeShoppingCart(w, r, clientId)
+	}
+}
+
+func handleShoppingCart(w http.ResponseWriter, r *http.Request) {
+	EnableCors(w, r)
+	SetContentTypeJson(w)
+	SetMaxAge(w, 0)
+	if r.Method == "GET" {
+		clientId := r.URL.Query().Get("clientId")
+		writeShoppingCart(w, r, clientId)
+	}
+}
+
+func writeShoppingCart(w http.ResponseWriter, r *http.Request, clientId string) {
+	w.Header().Set("Content-Type", "application/json")
+	discount := discounts[clientId]
+	if discount == 0 {
```

Comment:
why is this happening?

Reply:
refactored the whole section, should be hopefully clearer now